packages/expo-sensors/src/DeviceSensor.ts

```diff
@@ -16,48 +17,73 @@ export default class DeviceSensor<M> {
 
   constructor(nativeSensorModule: NativeSensorModule, nativeEventName: string) {
     this._nativeModule = nativeSensorModule;
+
     this._nativeEmitter = new EventEmitter(nativeSensorModule);
     this._nativeEventName = nativeEventName;
     this._listenerCount = 0;
   }
 
-  addListener(listener: Listener<M>): Subscription {
+  addListener = (listener: Listener<M>): Subscription => {
     const subscription = this._nativeEmitter.addListener(this._nativeEventName, listener);
     subscription.remove = () => this.removeSubscription(subscription);
     this._listenerCount++;
     return subscription;
-  }
+  };
 
-  hasListeners(): boolean {
+  hasListeners = (): boolean => {
     return this._listenerCount > 0;
-  }
+  };
 
-  getListenerCount(): number {
+  getListenerCount = (): number => {
     return this._listenerCount;
-  }
+  };
 
-  removeAllListeners(): void {
+  removeAllListeners = (): void => {
     this._listenerCount = 0;
     this._nativeEmitter.removeAllListeners(this._nativeEventName);
-  }
+  };
 
-  removeSubscription(subscription: Subscription): void {
+  removeSubscription = (subscription: Subscription): void => {
     this._listenerCount--;
     this._nativeEmitter.removeSubscription(subscription);
-  }
+  };
 
-  setUpdateInterval(intervalMs: number): void {
+  setUpdateInterval = (intervalMs: number): void => {
     if (!this._nativeModule.setUpdateInterval) {
       console.warn(`expo-sensors: setUpdateInterval() is not supported on ${Platform.OS}`);
     } else {
       this._nativeModule.setUpdateInterval(intervalMs);
     }
-  }
-  async isAvailableAsync(): Promise<boolean> {
+  };
+
+  isAvailableAsync = async (): Promise<boolean> => {
     if (!this._nativeModule.isAvailableAsync) {
       return false;
     } else {
       return await this._nativeModule.isAvailableAsync();
     }
-  }
+  };
+
+  getPermissionsAsync = async (): Promise<PermissionResponse> => {
+    if (!this._nativeModule.getPermissionsAsync) {
+      return defaultPermissionsResponse;
```

Comment:
```suggestion
    if (!this._nativeModule.getPermissionsAsync) {
      throw new UnavailabilityError('expo-sensors', 'getPermissionsAsync');
```
Aren't we following `UnavailableError`-scenario if native method is not exposed? 🤔

Reply:
i think this is a matter of convenience - in these cases, permissions are always available and so we can consider them granted. we could throw an error but then people would need to explicitly gate calls to this method based on platform, or handle the exceptions